numpy/core/src/multiarray/number.c

```diff
@@ -943,6 +942,8 @@ NPY_NO_EXPORT PyNumberMethods array_as_number = {
     .nb_inplace_floor_divide = (binaryfunc)array_inplace_floor_divide,
     .nb_inplace_true_divide = (binaryfunc)array_inplace_true_divide,
 
+    .nb_index = (unaryfunc)array_index,
+
```

Comment:
Thanks.  This is the main example where the reordering decreases readability (next one also a bit, why not group alloc/dealloc).
Since it is the only one, I am happy if we merge this with the hope it makes a C++ introduction either somewhere at some point.

Reply:
Readability is certainly an issue, but the style here was copied from the source of the field names: https://github.com/python/cpython/blob/main/Include/cpython/object.h#L95

Reply:
Well, definitions must preserve order for ABI complience.  But, the main file where this hurts readability a bit is probably also one of the ones higher up on the C++ conversion priority list (when it happens), so lets put this in.

Thanks @juztamau5.